src/solver/utils/ortools_utils.cpp

```diff
@@ -356,30 +358,29 @@ std::string availableOrToolsSolversString()
 
 static std::optional<std::string> translateSolverName(const std::string& solverName, bool isMip)
 {
-    try
+    if (!OrtoolsUtils::solverMap.contains(solverName))
     {
-        if (isMip)
-        {
-            return OrtoolsUtils::solverMap.at(solverName).MIPSolverName;
-        }
-        else
-        {
-            return OrtoolsUtils::solverMap.at(solverName).LPSolverName;
-        }
+        return {};
     }
-    catch (const std::out_of_range&)
+    auto names = OrtoolsUtils::solverMap.at(solverName);
+    if (isMip)
     {
-        return {};
+        return names.MIPSolverName.empty() ? std::optional<std::string>() : names.MIPSolverName;
+    }
+    else
+    {
+        return names.LPSolverName.empty() ? std::optional<std::string>() : names.LPSolverName;
     }
 }
 
 MPSolver* MPSolverFactory(const bool isMip, const std::string& solverName)
 {
-    const std::string notFound = "Solver " + solverName + " not found";
+    const std::string notFound = "Solver " + solverName + " was not found or does not support "
+                                 + (isMip ? "MIP" : "LP") + " problems";
     const std::invalid_argument except(notFound);
```

Comment:
Move in the `if(!internalSolverName.has_value())` block, since this should not happen very often there is no need to construct objects "just in case something goes wrong"